kotlin-android/app/src/main/java/io/realm/examples/objectserver/model/ListAdapter.kt

```diff
@@ -0,0 +1,98 @@
+package io.realm.examples.objectserver.model
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.PopupMenu
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import io.realm.OrderedRealmCollection
+import io.realm.Realm
+import io.realm.RealmRecyclerViewAdapter
+import io.realm.examples.objectserver.R
+import io.realm.kotlin.where
+import org.bson.types.ObjectId
+
+
+internal class ListAdapter(data: OrderedRealmCollection<Task>) : RealmRecyclerViewAdapter<Task, ListAdapter.MyViewHolder?>(data, true) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val itemView: View = LayoutInflater.from(parent.context).inflate(R.layout.task_view, parent, false)
+        return MyViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        val obj: Task? = getItem(position)
+        holder.data = obj
+        holder.name.text = obj?.name
+        holder.status.text = obj?.status
+
+        // multiselect popup to control status
+        holder.itemView.setOnClickListener { item -> run {
+            val popup = PopupMenu(holder.itemView.context, holder.menu)
+            popup.inflate(R.menu.task_options_menu)
+            popup.setOnMenuItemClickListener(PopupMenu.OnMenuItemClickListener { item: MenuItem? ->
+                val status: String = when (item!!.itemId) {
+                    R.id.menu1 -> {
```

Comment:
On the other tutorials we were selective and only presented possible options, i.e. any status other than the current one.

Reply:
Unfortunately with XML menu layouts that isn't possible. I'll see what I can do with programmatic options though.